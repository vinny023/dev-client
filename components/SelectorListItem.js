import React from 'react'
import {Text, TouchableOpacity } from 'react-native'
import { useNavigation } from '@react-navigation/native';


export default function SelectorListItem({ screen, item }) {
    const navigation = useNavigation();

    if (screen==="OrderDetailScreen") {
        const order = item
        return (
            <TouchableOpacity 
                style={{padding: 20, backgroundColor: '#FDD7E4', alignSelf: 'stretch',  textAlign: 'center',}}
                onPress={() => navigation.navigate(screen, order)}
            >
            <Text>{order.supplier} - {order.estimatedDeliveryDate}</Text>            
            </TouchableOpacity>
         )
    } 
    
    
}