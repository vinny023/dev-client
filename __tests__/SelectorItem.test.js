import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {Text, View} from 'react-native'
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';
import {create} from 'react-test-renderer'
// import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import SelectorListItem from '../components/SelectorListItem'

import {SelectorListItemTestProps} from './setup/TestConfig'

describe('PlusMinus', () => {

    const wrapper = shallow(<SelectorListItem/>)
    
    test('renders the component', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    // test('renders the component with correct props', () => {
      
    // })

    // test('navigates to the correct screen and passes the correct props', () => {

    // })      

 })