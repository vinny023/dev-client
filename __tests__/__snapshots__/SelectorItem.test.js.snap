// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusMinus renders the component 1`] = `
<Button
  onPress={[Function]}
  title="Go To undefined"
/>
`;
